### PR TITLE
Netwrix Auditor intake

### PR DESCRIPTION
**Pull request by m-242**
This is a netwrix intake that leverages the [CEF export addon for netwrix](https://www.netwrix.com/generic_siem_integration.html).

I have a wiki page describing setup (in french), in case it may be helpful to you.

I haven't been able to test all the log kinds provided by Netwrix, as I don't use all product they support, however, Exchange Logs, Sharepoint logs and logon activity logs are parsed.

### CODE REVIEW

File: Netwrix/netwrix_auditor/ingest/parser.yml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      - set:
+          file.path: '{{ cef_parser.message.filePath.split("/")[3:] |join("/") }}'
+          event.action: '{{ cef_parser.message.Name | lower }}'
+        filter: ''
+        name: set
+  logon-event-setter:
+    actions:
+      - set:
+          event.reason: '{{ cef_parser.message.msg }}'
+          server.address: '{{ cef_parser.message.shost }}'
```
Comment (squioc) [inline Netwrix/netwrix_auditor/ingest/parser.yml:59]: To preserve the semantics of `shost`, I suggest using `source.domain`:
```suggestion
          source.domain: '{{ cef_parser.message.shost }}'
```

File: Netwrix/netwrix_auditor/ingest/parser.yml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    filter: '{{ cef_parser.message.DeviceProduct == "Exchange Online" }}'
+    external: null
+stages:
+  general-setter:
+    actions:
+      - set:
+          event.kind: '{{ cef_parser.message.DeviceProduct }}'
+          event.category: '{{ cef_parser.message.DeviceEventClassID |lower }}'
+          event.severity: '{{ cef_parser.message.Severity }}'
+          event.start: '{{ cef_parser.message.start }}'
```
Comment (squioc) [inline Netwrix/netwrix_auditor/ingest/parser.yml:45]: You can also add information about the observer:
```suggestion
          event.start: '{{ cef_parser.message.start }}'
           observer.vendor: '{{parsed_event.message.DeviceVendor}}'
          observer.product: '{{parsed_event.message.DeviceProduct}}'
          observer.version: '{{parsed_event.message.DeviceVersion}}'
```

File: Netwrix/netwrix_auditor/ingest/parser.yml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  - name: exchange-event-setter
+    description: ''
+    filter: '{{ cef_parser.message.DeviceProduct == "Exchange Online" }}'
+    external: null
+stages:
+  general-setter:
+    actions:
+      - set:
+          event.kind: '{{ cef_parser.message.DeviceProduct }}'
+          event.category: '{{ cef_parser.message.DeviceEventClassID |lower }}'
```
Comment (squioc) [inline Netwrix/netwrix_auditor/ingest/parser.yml:43]: `event.code` is more suitable for this information
```suggestion
          event.code: '{{ cef_parser.message.DeviceEventClassID |lower }}'
```
<br/>

`event.category` has restricted values (see [ECS specification](https://www.elastic.co/guide/en/ecs/8.4/ecs-allowed-values-event-category.html#ecs-event-category-configuration)). I suggest for `event.category` to use `['configuration']` and for `event.type`, to use `['change']` according the tests.

File: Netwrix/netwrix_auditor/ingest/parser.yml [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    external: null
+  - name: exchange-event-setter
+    description: ''
+    filter: '{{ cef_parser.message.DeviceProduct == "Exchange Online" }}'
+    external: null
+stages:
+  general-setter:
+    actions:
+      - set:
+          event.kind: '{{ cef_parser.message.DeviceProduct }}'
```
Comment (squioc) [inline Netwrix/netwrix_auditor/ingest/parser.yml:42]: The value for `event.kind` are [restricted](https://www.elastic.co/guide/en/ecs/8.4/ecs-allowed-values-event-kind.html). I suggest to use `event` as value.
```suggestion
          event.kind: 'event'
```

**Review by squioc — CHANGES_REQUESTED** [review]
Thanks for your work.
Here, some improvements for the parser.

**Comment by m-242** [comment]
Thanks for your review.
I applied your changes, but we are losing some information now that the data that was in `event.kind` isn't put in a field.
I need it for the smart descriptions, would it be a problem to have it in another field ?

File: Netwrix/netwrix_auditor/ingest/parser.yml [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    filter: '{{ cef_parser.message.DeviceProduct == "Exchange Online" }}'
+    external: null
+stages:
+  general-setter:
+    actions:
+      - set:
+          event.kind: 'event'
+          event.code: '{{ cef_parser.message.DeviceEventClassID |lower }}'
+          event.severity: '{{ cef_parser.message.Severity }}'
+          event.start: '{{ cef_parser.message.start }}'
```
Comment (squioc) [inline Netwrix/netwrix_auditor/ingest/parser.yml:45]: I think you should use the stage date to parse this date in the message as [`event.start`](https://www.elastic.co/guide/en/ecs/8.4/ecs-event.html#field-event-start) accept date type (ie [RFC3339](https://www.rfc-editor.org/rfc/rfc3339#page-10) formatted date)

File: Netwrix/netwrix_auditor/ingest/parser.yml [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+          observer.vendor: '{{ cef_parser.message.DeviceVendor}}'
+          observer.product: '{{ cef_parser.message.DeviceProduct}}'
+          observer.version: '{{ cef_parser.message.DeviceVersion}}'
+        filter: ''
+        name: set
+  document-event-setter:
+    actions:
+      - set:
+          file.path: '{{ cef_parser.message.filePath.split("/")[3:] |join("/") }}'
```
Comment (squioc) [inline Netwrix/netwrix_auditor/ingest/parser.yml:54]: You can also provide `file.name` and `file.directory`:
```suggestion
          file.path: '{{ cef_parser.message.filePath.split("/")[3:] |join("/") }}'
          file.directory: '{{ cef_parser.message.filePath.split("/")[3:] |join("/") | dirname }}'
          file.name: '{{ cef_parser.message.filePath | basename }}'
```

**Review by squioc — APPROVED** [review]
Looks good. Thanks for the contribution

**Pull request merged**